File: example/old/sigmoid.py
import numpy as np

class sigmoid:
  def __repr__(self):
    return 'sigmoid()'

  def forward(self, input):
    return 1 / (1 + np.exp(-input))
    
  def backward(self, input):
    # return input * (1 - input) # this is when the input is already passed through the sigmoid function, the real derivative is below
    return np.exp(-input) / np.square(1 + np.exp(-input))

File: aiinpyc/sigmoid.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

double *sigmoid(double *x)
{
  for (int i = 0; i < 4; i++)
  {
    x[i] = 1 / (1 + exp(-x[i]));
  }
  return x;
}

int main()
{
  double input[4] = {1, 2, 3, 4};
  double *output = sigmoid(input);

  for (int i = 0; i < 4; i++)
  {
    printf("%f\n", output[i]);
  }
}